README.md

```diff
@@ -202,41 +202,262 @@ There are situations where you want to store some compiler settings in a config
+
+The object structure is as follows:
+
+```typescript
+{
+    /**
+     * A glob pattern string or file path, or an array of glob pattern strings and/or file paths.
+     * These can be relative paths or absolute paths.
+     * All non-absolute paths are resolved relative to the rootDir
+     */
+    src: Array<string | string[]>;
+    /**
+     * The relative path to the location in the output folder where the files should be placed,
+     * relative to the root of the output folder
+     */
+    dest: string | undefined
+}
+```
+
+If `src` is a non-glob path to a single file, then `dest` should include the filename and extension. For example:
+
+```jsonc
+{ "src": "lib/Promise/promise.brs", "dest": "source/promise.brs" }
+```
+
+If `src` is a glob pattern, then dest should be a path to the folder in the output directory. For example:
+
+```jsonc
+{ "src": "lib/*.brs", "dest": "source/lib" }
+```
+
+If `src` is a path to a folder, it will be ignored. If you want to copy a folder and its contents, use the glob syntax. The following example will copy all files from the lib/vendor folder recursively:
+
+```jsonc
+{ "src": "lib/vendor/**/*", "dest": "vendor" }
+```
+
+If `dest` is not specified it will default to the root of the output folder.
+
+An example of combining regular and advanced file patterns:
+
+```jsonc
+{
+    "rootDir": "C:/projects/CatVideoPlayer",
+    "files": [
+        "source/main.brs",
+        {
+          "src": "../common/promise.brs",
+          "dest": "source/common"
+        }
+    ]
+}
+```
+
+##### File collision handling
+
+Because file entries are processed in order you can override a file by specifying an alternative later in the files array.
+
+For example, if you have a base project and a child project that wants to override specific files, you could do the following:
+
+```jsonc
+{
+    "files": [
+        {
+            //copy all files from the base project
+            "src": "../BaseProject/**/*"
+        },
+        // Override "../BaseProject/themes/theme.brs"
+        // with "${rootDir}/themes/theme.brs"
+        "themes/theme.brs"
+    ]
+}
+```
+
+#### `outFile`
+
+Type: `string`
+
+The path (including filename) where the output file should be placed. Defaults to `"./out/${WORKSPACE_FOLDER_NAME}.zip"`.
+
+#### `createPackage`
+
+Type: `boolean`
+
+Causes the build to create a zip package. Defaults to `true`. This setting is ignored when `deploy` is enabled.
+
+#### `watch`
+
+Type: `boolean`
+
+If `true`, the server will keep running and will watch and recompile on every file change. Defaults to `false`.
+
+#### `deploy`
+
+Type: `boolean`
+
+If `true`, after a successful build, the project will be deployed to the Roku specified in host. Defaults to `false`. If this field is set to `true`, then the `host`, `username`, and `password` fields must be set as well.
```

Comment:
I haven't tested whether `username` is required for `deploy` yet, so this is an assumption on my part.

Reply:
username is not required. Roku devices do not allow developers to _change_ the username, so it's always `rokudev`. Since that's the case, it's not required because our tools can default to `"rokudev"`.

Reply:
Thanks, updated in the new version.